### PR TITLE
Musark 1000

### PR DESCRIPTION
**Pull request by rituvesh**

### CODE REVIEW

File: src/modules/analysis/AnalysisViewComponent.js [hunk 1]
```diff
@@ -81,13 +83,13 @@ const getObjectsValue = store => {
 };
 const getValue = (store, field) => {
   if (store.analysis) {
-    if (
+ /*   if (
```
Comment (jarlah) [inline src/modules/analysis/AnalysisViewComponent.js]: remove

File: src/modules/analysis/AnalysisViewComponent.js [hunk 2]
```diff
@@ -343,9 +345,13 @@ const AnalysisView = ({ store }: Props) => (
       </FormGroup>
     </Form>
     <NewLine />
-    <button>
-      close
-    </button>
+    <SaveCancel
+      onClickSave={(e) => {
+      e.preventDefault();
```
Comment (jarlah) [inline src/modules/analysis/AnalysisViewComponent.js]: formatting? should be at least 2 space indented

File: src/modules/analysis/__tests__/analysisAddComponent-test.js [hunk 3]
```diff
@@ -56,18 +56,6 @@ const form = fieldsArray.reduce(
 );
 
 describe('AnalysisAddComponent', () => {
-  it('saveAnalysisEventLocal should call saveAnalysisEvent', () => {
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisAddComponent-test.js]: why remove?

File: src/modules/analysis/__tests__/analysisEditComponent-test.js [hunk 4]
```diff
@@ -60,18 +57,6 @@ const form = fieldsArray.reduce(
 );
 
 describe('AnalysisEditComponent', () => {
-  it('editAnalysisEventLocal should call editAnalysisEvent', () => {
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisEditComponent-test.js]: why remove?

File: src/modules/analysis/AnalysisViewComponent.js [hunk 5]
```diff
@@ -152,18 +154,18 @@ const AnalysisView = ({ store }: Props) => (
       <Col mdOffset={1} md={5}>
         <Table responsive>
           <thead>
-            <tr>
-              <th>Museumsnr</th>
-              <th>Unr</th>
-              <th>Term/artsnavn</th>
-            </tr>
+          <tr>
```
Comment (jarlah) [inline src/modules/analysis/AnalysisViewComponent.js]: definitely not formatted ;)

File: src/modules/analysis/AnalysisViewComponent.js [hunk 6]
```diff
@@ -14,29 +14,31 @@ import {
   Panel
 } from 'react-bootstrap';
 import FontAwesome from 'react-fontawesome';
-
-type Props = { store: any };
+import {SaveCancel} from '../../components/formfields/index';
+import {hashHistory} from 'react-router';
+import Config from '../../config';
+type Props = { store: any, appSession: any, params: any };
```
Comment (jarlah) [inline src/modules/analysis/AnalysisViewComponent.js]: i think you should iron out the properties you are using on these three Props fields.

**Review by jarlah — CHANGES_REQUESTED** [review]
A couple of comments/requirements

File: src/modules/analysis/__tests__/analysisAddComponent-test.js (hunk 3 above)
Comment (rituvesh) [reply src/modules/analysis/__tests__/analysisAddComponent-test.js]: showing error for .then. I will try to add it back

File: src/modules/analysis/__tests__/analysisEditComponent-test.js (hunk 4 above)
Comment (rituvesh) [reply src/modules/analysis/__tests__/analysisEditComponent-test.js]: same as above!

File: src/modules/analysis/AnalysisViewComponent.js (hunk 6 above)
Comment (rituvesh) [reply src/modules/analysis/AnalysisViewComponent.js]: Done!

File: src/modules/analysis/AnalysisViewComponent.js (hunk 1 above)
Comment (rituvesh) [reply src/modules/analysis/AnalysisViewComponent.js]: done!

File: src/modules/analysis/AnalysisViewComponent.js (hunk 5 above)
Comment (rituvesh) [reply src/modules/analysis/AnalysisViewComponent.js]: Done.

File: src/modules/analysis/AnalysisViewComponent.js (hunk 2 above)
Comment (rituvesh) [reply src/modules/analysis/AnalysisViewComponent.js]: Done.

File: src/modules/analysis/__tests__/analysisAddComponent-test.js [hunk 7]
```diff
@@ -55,17 +59,33 @@ const form = fieldsArray.reduce(
...
+    saveAnalysisEventLocal(appSession, form, store, saveAnalysisEvent)(1);
+    expect(saveAnalysisEvent.calledOnce).toBe(true);
+    expect(saveAnalysisEvent.getCall(0).args[0].museumId).toBe(99);
+    expect(saveAnalysisEvent.getCall(0).args[0].token).toBe('1234');
+    expect(saveAnalysisEvent.getCall(0).args[0].data.analysisTypeId).toBe(
+      'b15ee459-38c9-414f-8b54-7c6439b44d3d'
+    );
+  });
+
+  it('meh', done => {
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisAddComponent-test.js]: rename to something more valuable, i was just making a throwaway name :)

File: src/modules/analysis/__tests__/analysisAddComponent-test.js [hunk 8]
```diff
@@ -55,17 +59,33 @@ const form = fieldsArray.reduce(
...
+    const fakeGoTo = sinon.spy();
+    const fakeFn = () => Observable.of(null).toPromise();
+    const fn = goToAnalysis(fakeFn, appSession, fakeGoTo);
+    const analysisId = 2;
+    fn(analysisId).then(() => {
+      expect(fakeGoTo.calledOnce).toBe(true);
+      expect(fakeGoTo).toMatchSnapshot();
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisAddComponent-test.js]: remove this expect. instead add ```expect(fakeGoTo.getCall(0).args[0]).toBe('/api/whatever/museum');```

File: src/modules/analysis/__tests__/analysisEditComponent-test.js [hunk 9]
```diff
@@ -59,17 +63,33 @@ const form = fieldsArray.reduce(
...
+    editAnalysisEventLocal(appSession, form, store, editAnalysisEvent, param)(1);
+    expect(editAnalysisEvent.calledOnce).toBe(true);
+    expect(editAnalysisEvent.getCall(0).args[0].museumId).toBe(99);
+    expect(editAnalysisEvent.getCall(0).args[0].token).toBe('1234');
+    expect(editAnalysisEvent.getCall(0).args[0].data.analysisTypeId).toBe(
+      'b15ee459-38c9-414f-8b54-7c6439b44d3d'
+    );
+  });
+
+  it('meh', done => {
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisEditComponent-test.js]: rename

File: src/modules/analysis/__tests__/analysisEditComponent-test.js [hunk 10]
```diff
@@ -59,17 +63,33 @@ const form = fieldsArray.reduce(
...
+    const fakeGoTo = sinon.spy();
+    const fakeFn = () => Observable.of(null).toPromise();
+    const fn = goToAnalysis(fakeFn, appSession, fakeGoTo);
+    const analysisId = 2;
+    fn(analysisId).then(() => {
+      expect(fakeGoTo.calledOnce).toBe(true);
+      expect(fakeGoTo).toMatchSnapshot();
```
Comment (jarlah) [inline src/modules/analysis/__tests__/analysisEditComponent-test.js]: same same

File: src/modules/analysis/__tests__/analysisAddComponent-test.js (hunk 8 above)
Comment (rituvesh) [reply src/modules/analysis/__tests__/analysisAddComponent-test.js]: Got below error on it.
Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

File: src/modules/analysis/__tests__/analysisAddComponent-test.js (hunk 7 above)
Comment (rituvesh) [reply src/modules/analysis/__tests__/analysisAddComponent-test.js]: Done.

File: src/modules/analysis/__tests__/analysisEditComponent-test.js (hunk 9 above)
Comment (rituvesh) [reply src/modules/analysis/__tests__/analysisEditComponent-test.js]: Done.

**Review by jarlah — APPROVED** [review]

**Pull request merged**